### DISCUSSION TITLE
How to include List of tables and List of figures when generating a word document?

### DISCUSSION THREAD

**Discussion by selvavm** [quarto-dev/quarto-cli#5945] [Q&A] [answered]
### Description

I would like to include List of tables and List of figures in a word document that is generated by Quarto. I am trying to convert ipynb and also tried qmd with `lot` and `lof` options but it does not generate. If I try the same to generate PDF, they are included in the PDF document. Is there a way to include this in word document?

Note: I am using reference-doc to embed styles, header, and footer.

**Comment by mcanouil** [comment]
Valid options for Word document are listed in <https://quarto.org/docs/reference/formats/docx.html>.
As you can see, there is no `lot` or `lof` options.

Word documents are not well supported by Pandoc thus by Quarto.
See <https://github.com/jgm/pandoc/issues/1965> for example.

I am afraid (as far as I know), there is no simple solution.
A possible solution could be to write a filter to do it.

**Reply by selvavm** [reply]
Thanks @mcanouil. Do you know any filter that does this or is there any pointer for creating the filter? Sorry I am new to pandoc and quarto. 

I looked in [Lua filter](https://quarto.org/docs/extensions/filters.html) which mentions filter can change the AST but I feel missing lot/lof is more of a writer issue. Please correct me if I am wrong.

**Reply by mcanouil** [reply]
In some aspect, it is a writer issue.
The Lua filter will basically have to write the Open XML code into the template document.

Unfortunately, I never tried to work with Word using any programming languages (Lua included).

**Reply by selvavm** [reply]
Thanks @mcanouil. I will try to figure it out

**Comment by cscheid** [comment, marked as answer]
@mcanouil is correct that we currently don't have support for list of tables and list of figures.

With that said, I plan on implementing this for 1.4. See #2138 and #4944 for broader context.

**Reply by mcanouil** [reply]
Thanks, I missed that part in the epic.

**Reply by selvavm** [reply]
Thanks @cscheid. Good to hear this is work in progress. Do you have a tentative timeline for 1.4 official release?

**Reply by cscheid** [reply]
No.

**Comment by olivroy** [comment]
FWIW, this feature will make it in pandoc 3.5 https://github.com/jgm/pandoc/pull/10029.

**Reply by mcanouil** [reply]
Thanks for sharing!
